src/Microsoft.AspNet.Mvc.Core/DefaultControllerTypeProvider.cs

```diff
@@ -76,17 +77,45 @@ protected internal virtual bool IsController([NotNull] TypeInfo typeInfo)
             {
                 return false;
             }
-            if (typeInfo.Name.Equals("Controller", StringComparison.OrdinalIgnoreCase))
+            if (typeInfo.Name.Equals(ControllerTypeName, StringComparison.OrdinalIgnoreCase))
```

Comment:
Not sure why this code was structured this way but I spoke to @yishaigalatzer and we decided to

1. Drop this check
2. Make the `Controller` type in Mvc abstract. 

This should get us the same-ish results with the guidance being if you want a base controller type named `Controller` but that isn't directly accessible, mark it abstract.